src/commands/music/index.js

```diff
@@ -0,0 +1,215 @@
+import Promise from 'bluebird';
+import R from 'ramda';
+import ytdl from 'ytdl-core';
+
+import { addSong, delSong, delSongs, getSong, getSongs, getNextSong } from '../../redis';
+import { time } from '../../helpers';
+import T from '../../translate';
+
+let timer;
+
+
+function vleave(client, evt, lang) {
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (info) {
+    info.voiceConnection.disconnect();
+    delSongs(evt.message.guild.id);
+    delSong(evt.message.guild.id);
+  } else {
+    return Promise.resolve(T('music_not_connected', lang));
+  }
+}
+
+function vjoin(client, evt, suffix, lang) {
+  if (!suffix) {
+    const channel = evt.message.guild.voiceChannels.filter(vc => vc.members.map(m => m.id).indexOf(evt.message.author.id) > -1);
+    if (channel.length > 0) {
+      channel[0].join();
+      timer = setTimeout(vleave, 1200000, client, evt);
+    } else {
+      return Promise.resolve(T('music_not_found', lang));
+    }
+  } else {
+    const channel = evt.message.guild.voiceChannels.find(c => c.name.toLowerCase() === suffix.toLowerCase());
+    if (channel) {
+      channel.join();
+      timer = setTimeout(vleave, 1200000, client, evt);
+    } else {
+      return Promise.resolve(T('music_not_found', lang));
+    }
+  }
+}
+
+function play(client, evt, music, lang) {
+  const info = client.VoiceConnections.getForGuild(evt.message.guild);
+  if (!info) return Promise.resolve(T('music_not_connected', lang));
+  const encoder = info.voiceConnection.createExternalEncoder({
+    type: 'ffmpeg',
+    realtime: true,
+    source: music.url,
+    outputArgs: ['-compression_level', 7]
+  });
+  clearTimeout(timer);
+  encoder.play();
+  encoder.once('end', () => {
+    delSong(evt.message.guild.id);
+    getSong(evt.message.guild.id, (err, song) => {
+      if (err) throw err;
+      if (song) {
+        play(client, evt, JSON.parse(song));
+      } else {
+        timer = setTimeout(vleave, 1200000, client, evt);
+        return Promise.resolve(T('music_queue_empty', lang));
+      }
+    });
+  });
+}
+
+function request(client, evt, suffix, lang) {
```

Comment:
This method is a bit hard to read without spacing between some functions. Spaces?  :D